src/modm/platform/can/atsam-fdcan/can_interrupt.cpp.in

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023, Raphael Lehmann
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+
+#include "can_{{ id }}.hpp"
+
+namespace modm::platform
+{
+
+// line 0 used as TX and error interrupt
+// generated on finished frame transmission and error state
+MODM_ISR(MCAN{{ id }}_INT0)
+{
+    // FIXME this does not work if the a non-default MessageRamConfig is used
+	const auto callback = Mcan{{ id }}<>::getErrorInterruptCallback();
```

Comment:
Any suggestion how to access the  "static-"class with unknown template parameters from the interrupt?

Reply:
You can use a regular value type for the message RAM config and pass it to the driver as a non-type template parameter. That can be stored in a variable and accessed from the interrupt.